storage/handler/writer.go

```diff
@@ -135,26 +124,57 @@ func (w *Writer) Write(stream storage.WriteService_WriteServer) error {
 	}
 }
 
+func (w *Writer) handleReplica(replica *storage.Replica) {
+	reader := streamIO.NewReader(replica.Data)
+	for !reader.Empty() {
```

Comment:
there are 2 ways to reduce the test miss here:
1. you can use `for !reader.Empty() && reader.Err() == nil`
2. check reader.Err() after reading all data, it is not necessary to check error each step, it's too hard to mock every errors for data corruption